src/types/timestamp.rs

```diff
@@ -311,15 +291,9 @@ impl Timestamp {
 
         // TODO: make Timestamp::fractional_seconds to be Mantissa when creating a Timestamp to get rid of below conversion
         // convert Option<&Mantissa> to &Mantissa
-        let m1 = match &self.fractional_seconds {
-            None => &Mantissa::Digits(0),
-            Some(m) => m,
-        };
+        let m1 = &self.fractional_seconds;
 
-        let m2 = match &other.fractional_seconds {
-            None => &Mantissa::Digits(0),
-            Some(m) => m,
-        };
+        let m2 = &other.fractional_seconds;
 
         // compare fractional seconds
         match (m1, m2) {
```

Comment:
I don't see `m1` and `m2` getting used outside the `match` line below, so maybe we could just simplify it to:

```rust
match (&self.fractional_seconds, &other.fractional_seconds) {
```
?